File: examples/kafka-quickstart/src/main/java/acme/Sender.java
package acme;

import java.util.concurrent.*;

import jakarta.enterprise.context.ApplicationScoped;

import org.eclipse.microprofile.reactive.messaging.Outgoing;

import io.smallrye.reactive.messaging.kafka.KafkaRecord;

@ApplicationScoped
public class Sender {

    private ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();

    @Outgoing("data")
    public CompletionStage<KafkaRecord<String, String>> send() {
        CompletableFuture<KafkaRecord<String, String>> future = new CompletableFuture<>();
        delay(() -> future.complete(KafkaRecord.of("kafka", "key", "hello from MicroProfile")));
        return future;
    }

    private void delay(Runnable runnable) {
        executor.schedule(runnable, 5, TimeUnit.SECONDS);
    }

}

File: examples/kafka-quickstart-kotlin/src/main/kotlin/acme/Sender.kt
package acme

import java.util.concurrent.*

import jakarta.enterprise.context.ApplicationScoped

import org.eclipse.microprofile.reactive.messaging.Outgoing

import io.smallrye.reactive.messaging.kafka.KafkaMessage
import io.smallrye.reactive.messaging.kafka.KafkaRecord

@ApplicationScoped
open class Sender {

    private val executor = Executors.newSingleThreadScheduledExecutor()

    @Outgoing("data")
    open fun send(): CompletionStage<KafkaRecord<String, String>> {
        val future = CompletableFuture<KafkaRecord<String, String>>()
        delay(Runnable { future.complete(KafkaRecord.of("kafka", "key", "hello from MicroProfile")) })
        return future
    }

    private fun delay(runnable: Runnable) {
        executor.schedule(runnable, 5, TimeUnit.SECONDS)
    }

}
